heroku_connect/admin.py

```diff
@@ -0,0 +1,139 @@
+from urllib.parse import parse_qs, urlencode, urlsplit, urlunsplit
+
+from django.contrib import admin
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+from django.utils.html import format_html
+
+from heroku_connect.models import (
+    TriggerLog, TriggerLogArchive, TriggerLogPermanent, TriggerLogState
+)
+
+
+def _replaced(__values, **__replacements):
+    """Replace elements in iterable with values from an alias dict, suppressing empty values.
+
+    Used to consistently enhance how certain fields are displayed in list and detail pages.
+    """
+    return tuple(o for o in (__replacements.get(name, name) for name in __values) if o)
+
+
+def _get_admin_route_name(model_or_instance):
+    """Get the base name of the admin route for a model or model instance.
+
+    For use with :func:`django.urls.reverse`, although it still needs the specific route suffix
+    appended, for example ``_changelist``.
+    """
+    model = model_or_instance if isinstance(model_or_instance, type) else type(model_or_instance)
+    content_type = ContentType.objects.get_for_model(model)
```

Comment:
why do you need the content type to build an admin url? You can get the app_label and model name from the model or an instance directly. `model=cls.__name__` `app_label=cls._meta.app_label`